astropy/config/tests/test_configs.py

```diff
@@ -37,26 +37,122 @@ def test_paths():
     assert "testpkg" in paths.get_cache_dir(rootname="testpkg")
 
 
+@pytest.fixture
+def isolated_setup(monkeypatch):
+    # ignore global state of the test session
+    monkeypatch.delenv("ASTROPY_CACHE_DIR", raising=False)
+    monkeypatch.delenv("ASTROPY_CONFIG_DIR", raising=False)
+    monkeypatch.delenv("XDG_CACHE_HOME", raising=False)
+    monkeypatch.delenv("XDG_CONFIG_HOME", raising=False)
+
+    monkeypatch.setattr(paths.set_temp_cache, "_temp_path", None)
+    monkeypatch.setattr(paths.set_temp_config, "_temp_path", None)
```

Comment:
Is this thread safe? Don't you need to use locks explicitly?

Reply:
It's not. The current implementation is just meant for code sharing. I'll add locking now but thread safety will only be guaranteed in between users of this fixture.

Reply:
> ...thread safety will only be guaranteed in between users of this fixture.

That's why it would be better to make the code inherently type safe, but if that's too complicated then the fixture will have to do.